azuredata/src/main/java/com/azure/data/service/DocumentClient.kt

```diff
@@ -710,6 +711,9 @@ class DocumentClient(private val baseUri: ResourceUri, key: String, keyType: Tok
     // create
     private fun <T : Resource> create(resource: T, resourceUri: UrlLink, resourceType: ResourceType, additionalHeaders: Headers? = null, callback: (ResourceResponse<T>) -> Unit) {
 
+        if (!isNetworkAvailable(ContextProvider.appContext)){
+            return callback(ResourceResponse(DataError.fromType(ErrorType.NetworkUnavailable)))
+        }
```

Comment:
We wouldn't want to check reachability when making calls here.  Desired behavior being discussed in Slack: https://azuremobile.slack.com/archives/C9X9NUP9T/p1523370455000158